hw9/code/intro_pandas.py

```diff
@@ -0,0 +1,149 @@
+pokemons_df.info()
+
+# Renaming columns
+new_columns = {'#': 'Id', 'Type 1': 'Type_1', 'Type 2': 'Type_2', 'Sp. Atk': 'Sp_atk', 'Sp. Def': 'Sp_def'}
+pokemons_df.rename(columns=new_columns, inplace=True)
+
+# Checking for missing values
+print('Не стоит беспокоиться об NA') if pokemons_df.count().min() == pokemons_df.shape[0] else print('Oops!')
+
+plt.figure(figsize=(10, 8))
+plt.title('Heatmap of NA values in dataframe', fontsize=18)
+sns.heatmap(pokemons_df.isna())
+plt.xticks(fontsize=14)
+
+print('\tПеременная\tПроцент NA')
+print('---------------------------------------')
+for col in pokemons_df.columns:
+    percentage = pokemons_df[col].isna().mean()*100
+    if percentage == 0:
+        percentage = int(percentage)
+    else:
+        percentage = round(percentage, 2)
+    print('{:>14} {:>14} %'.format(col, percentage))
+
+# Plotting Type 1 and Type 2 classes
+plt.figure(figsize=(15, 8))
+plt.subplot(1, 2, 1)
+sns.countplot(pokemons_df['Type_1'])
+plt.title('Type 1 classes', fontsize=18)
+plt.xlabel('Type 1', fontsize=14)
+plt.ylabel('Count', fontsize=14)
+plt.xticks(rotation=90)
+
+plt.subplot(1, 2, 2)
+sns.countplot(pokemons_df['Type_2'])
+plt.title('Type 2 classes', fontsize=18)
+plt.xlabel('Type 2', fontsize=14)
+plt.ylabel('Count', fontsize=14)
+plt.xticks(rotation=90)
+
+# Plotting Generation and Legendary classes
+plt.figure(figsize=(15, 8))
+plt.subplot(1, 2, 1)
+sns.countplot(pokemons_df.Generation)
+plt.xlabel('Generation', fontsize=14)
+plt.ylabel('count', fontsize=14)
+plt.xticks(fontsize=14)
+plt.yticks(fontsize=14)
+
+plt.subplot(1, 2, 2)
+sns.countplot(pokemons_df.Legendary)
+plt.xlabel('Legendary', fontsize=14)
+plt.ylabel('count', fontsize=14)
+plt.xticks(fontsize=14)
+plt.yticks(fontsize=14)
+
+# Plotting pair plots for numeric features
+numeric_cols = ['Total', 'HP', 'Attack', 'Defense', 'Sp_atk', 'Sp_def', 'Speed']
+sns.pairplot(pokemons_df[numeric_cols])
+
+# Plotting boxplots for numeric features
+plt.figure(figsize=(18, 10))
+for i, col in enumerate(numeric_cols):
+    plt.subplot(2, 4, i+1)
+    sns.boxplot(pokemons_df[col])
+
+# Plotting Pearson correlation heatmap for numeric features
+plt.figure(figsize=(10, 8))
+sns.heatmap(pokemons_df[numeric_cols].corr(), annot=True)
+plt.title('Pearson correlation', fontsize=18)
+plt.xticks(fontsize=14)
+plt.yticks(fontsize=14)
+
+
+# functions for reading gff and bed formats
+def read_gff(path_to_file):
+    rrna_annotation_df = pd.read_csv(path_to_file, header=1, sep='\t')
+    rrna_annotation_df = pd.DataFrame(np.r_[np.array(rrna_annotation_df.columns)[np.newaxis, :],
+                                            rrna_annotation_df.values])
+    rrna_annotation_df.columns = ['chromosome', 'source', 'type', 'start', 'end',
+                                  'score', 'strand', 'phase', 'attributes']
+    rrna_annotation_df[['start', 'end']] = rrna_annotation_df[['start', 'end']].astype(int)
+    rrna_annotation_df['score'] = rrna_annotation_df['score'].astype(float)
```

Comment:
Я не совсем понял к чему весь этот огород с индексами. Чтобы убрать комменты?
```suggestion
    rrna_annotation_df = pd.read_table(path_to_file, header=None, names=['chromosome', 'source', 'type', 'start', 'end',  'score', 'strand', 'phase', 'attributes'], comment="#")
    rrna_annotation_df[['start', 'end']] = rrna_annotation_df[['start', 'end']].astype(int)
    rrna_annotation_df['score'] = rrna_annotation_df['score'].astype(float)
```